docs/manual-EN/3.build-develop-and-administration/1.build/1.build-source-code.md

```diff
@@ -1,244 +1,200 @@
 
-- If no errors are shown
+#### CMAKE_BUILD_TYPE
+There are a few building types supported:
+ * `Debug`, to build with debuginfo but without optimization, which is by default
+ * `Release`, to build with optimization but without debuginfo
+ * `RelWithDebInfo`, to build with optimization AND debuginfo
+ * `MinSizeRel`, to build with optimizations for code size
 
-    ```bash
-    [100%] Built target ....
-    ```
-
-    **Congratulations! Compile successfully...**
-
-- You can see there are four folders in the the installation directory **/usr/local/nebula**: **etc/**, **bin/**, **scripts/**, **share/**.
-
-    ```bash
-    [root@centos7.5 nebula]# ls /usr/local/nebula/
-    bin  etc  scripts  share
-    ```
-
-**Now, you can start Nebula Graph!**
-
-#### Start Nebula Graph Services
-
-After the above installation is completed, you can start the Nebula Graph services. The default directory of Nebula Graph is `/usr/local/nebula`.
+#### CMAKE_INSTALL_PREFIX
+This option is to specify the location where the service modules, scripts, configuration files and tools are installed when `make install` is performed. It is set to `/usr/local/nebula` by default.
 
+#### CMAKE_CXX_COMPILER
+Normally, CMake will figure out and locate an applicable C++ compiler for us. But if your compiler installation is not at the standard location, or if you want to use a different one, you have to specify it explicitly, like:
 ```bash
-bash> cd /usr/local/nebula
+$ cmake -DCMAKE_C_COMPILER=/path/to/gcc/bin/gcc -DCMAKE_CXX_COMPILER=/path/to/gcc/bin/g++ ..
+$ cmake -DCMAKE_C_COMPILER=/path/to/clang/bin/clang -DCMAKE_CXX_COMPILER=/path/to/clang/bin/clang++ ..
 ```
 
-Rename the configuration files for Nebula Graph services.
-
-```bash
-bash> cp etc/nebula-graphd.conf.default etc/nebula-graphd.conf
-bash> cp etc/nebula-metad.conf.default etc/nebula-metad.conf
-bash> cp etc/nebula-storaged.conf.default etc/nebula-storaged.conf
-```
+#### ENABLE_CCACHE
+This option is to enable the use of `ccache`, which is for speeding up the compilation during daily development.
 
-Start Nebula Graph services.
+By default, **Nebula Graph** will take advantage of `ccache` if it's found. So you don't have to enable it for yourself.
 
-```bash
-bash> ./scripts/nebula.service start all
-bash> ./bin/nebula -u user -p password --port 3699 --addr="127.0.0.1"
-nebula> SHOW HOSTS;
-```
+If you want to disable `ccache`, it might be not enough to just turn `ENABLE_CCACHE` off. Since on some platforms, the `ccache` installation hooks up or precedes the compiler. For such an case, you have to set an evnironment variable `export CCACHE_DISABLE=true`, or add a line `disable=true` to `~/.ccache/ccache.conf`. We may do this for you automatically in future.
 
-## Trouble Shooting
+Please see the [official documentation](https://ccache.dev/manual/3.7.6.html) for more details on it.
```

Comment:
For details, refer to [official documentation](https://ccache.dev/manual/3.7.6.html).